server/lib/data_import/my_anime_list/extractor/character.rb

```diff
@@ -0,0 +1,107 @@
+module DataImport
+  class MyAnimeList
+    module Extractor
+      class Character
+        include Enumerable
+
+        attr_reader :dom, :external_id
+
+        def initialize(html, external_id)
+          @dom = Nokogiri::HTML(html)
+          @external_id = external_id
+        end
+
+        def name
+          # line 473 in guts-character.html
+          options = ['Animeography', 'Mangaography', 'Voice Actors']
+          dom.css('.normal_header').map do |row|
+            next if  options.include?(row.content.strip)
+
+            english = row.children.first.content.strip
+            # japanese = row.children.at_css('small').content.strip
+
+            return english
+          end
+        end
+
+        def image_url
+          # line 319 in guts-character.html
+          {
+            image: dom.at_css("a[href='/character/#{external_id}/#{name.tr(' ', '_')}/pictures'] img").attr('src')
+          }
+        end
+
+        def description
+          dom.css('#content > table >tr > td:nth-child(2)').children.take_while { |x|
+            !x.text.include? 'Voice Actor'
+          }.reject { |x|
+            x['class'] == 'normal_header' || x['id'] == 'horiznav_nav' ||
+            x['itemtype'] == 'http://schema.org/BreadcrumbList'
+          }.map { |x|
+            reg_check = /\A(.+(?=:)|.{0,30})\z/i
+
+            if x.attributes['class'].try(:value) == "spoiler"
+              if x.previous.previous.previous.present? && !(x.previous.previous.previous.content =~ reg_check)
```

Comment:
checking where the spoiler is actually related to: (is it related to the actual description or part of attributes which we don't care about). 
Guts character spoiler is related to his weight which I am throwing away anyway so we don't need that spoiler. 
Shichiroji character spoiler is related to the description so we need to keep it.